blog/2020-06-angular10/README.md

```diff
@@ -0,0 +1,157 @@
+
+## Update der TypeScript Umgebung
+
+### TypeScript 3.9
+
+Mit Angular 10 unterstützt Angular nun [TypeScript in der Version 3.9](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-9.html).
+
+### tslib 2.0.0
+
+Weiterhin wurde die Abhängigkeit zur TypeScript Bibliothek _tslib_ auf die Version _2.0.0_ angehoben.
+Sofern Sie ein existierendes Angular Projekt mit abhängigen Packages haben, die eine `peerDependency` von tslib 1.x.x erfordern, werden Sie bei der Ausführung von `ng update` wahrscheinlich folgende Fehlermeldung erhalten:
+
+```
+Package "my-package" has an incompatible peer dependency to "tslib" (requires "^1.10.0", would install "2.0.0").
+✖ Migration failed: Incompatible peer dependencies found.
+Peer dependency warnings when installing dependencies means that those dependencies might not work correctly together.
+You can use the '--force' option to ignore incompatible peer dependencies and instead address these warnings later.
+```
+
+Um dieses Problem zu beheben können Sie in vielen Fällen die Fehlermeldung ignorieren und einfach `ng update` mit dem Flag `--force` ausführen.
+Hinterher sollten Sie jedoch zur Sicherheit noch einmal genauestens prüfen, ob ihre Anwendung und ihrer Tests korrekt funktionieren.
+
+Ist Ihnen dieser Weg zu reisikoreich, empfehlen wir Ihnen sich etwas zu gedulden, bis die zu nutzende Bibliothek `tslib` 2.0.0 unterstützt.
+Prüfen Sie ggf. beim Provider der Bibliothek ob bereits ein Issue zu diesem Thema eröffnet wurde oder stellen Sie ein entsprechendes Issue für die Unterstützung von Angular 10 / tslib 2.0.0 ein.
+
+### TSLint 6.0.0
+
+Ab Angular 10 wird TSLint in der Version 6.0.0 unterstützt.
+Sämtliche Änderungen an der Datei `tslint.json` führen die Angular Schematics bei der Ausführung von `ng update` automatisch durch.
+
+
+## Browserslist
+
+Mit Angular 10 wurde die Standardkonfiguration für unterstütze Browser etwas aufgeräumt.
+Weiterhin hat sich der Dateiname von `browserslist` zu `.browserslistrc` geändert.
+
+Standardmäßig wird nun vor allem die letzte oder die letzten beiden Version der meist genutzten Browser unterstützt.
+Wenn Sie das tool `browserslist` über `npx` ausführen, erhalten Sie eine detaillierte Auflistung der unterstützten Browser bei der aktuellen Konfiguration der Datei `.browserslistrc`.
+
+```bash
+$ npx browserslist
+chrome 83
+edge 83
+edge 81
+firefox 77
+firefox 68
+ios_saf 13.4-13.5
+ios_saf 13.3
+ios_saf 13.2
+ios_saf 13.0-13.1
+ios_saf 12.2-12.4
+ios_saf 12.0-12.1
+safari 13.1
+safari 13
+safari 12.1
+safari 12
+```
+
+Sie können die [Werte hier selbstverständlich anpassen](https://github.com/browserslist/browserslist).
+Wollen Sie beispielweise den Internet Explorer 11 unterstützen, fügen Sie einfach `IE 11` zur Datei hinzu.
+Angular berücksichtigt die Einstellungen beim Build und erstellt beispielsweise Bundles mit ES5 Unterstützung, sofern sich ein Browser unter den Zielen befindet, die einen neueren JavaScript-Standard nicht unterstützen.
```

Comment:
Sind wir uns da sicher? In der Vergangenheit war es ja vor allem nur CSS-Vendor-Prefixe. Sollte das wirklich so sein, wäre dies natürlich sehr cool.

Reply:
Stand so im offiziellen Angular Changelog:


„ This has the side effect of disabling ES5 builds by default for new projects. To enable ES5 builds and differential loading for browsers that require it (such as IE or UC Browser), simply add the browsers you need to support in the .browserslistrc file.“